Rocket/Rocket/RocketsList/RocketsListView.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  RocketsListView.swift
+//  Rocket
+//
+//  Created by Adela Mišicáková on 11.07.2022.
+//
+
+import SwiftUI
+import ComposableArchitecture
+import ComposablePresentation
+
+//MARK: - State
+
+struct AppState: Equatable {
+    var detailState: DetailState? = nil
+    var fetchingState = FetchingState.na
+    var rockets = [Rocket]()
+    var details = IdentifiedArrayOf<DetailState>()
+    var alert: AlertState<AppAction>?
+    
+    enum FetchingState: Equatable {
+        case na
+        case loading
+        case success([Rocket])
+        case error(String)
+    }
+}
+
+//MARK: - Action
+
+enum AppAction: Equatable {
+    case retry
+    case getRockets
+    case rocketsResponse(Result<[Rocket], RocketsManager.Failure>)
+    case detail(id: UUID, action: DetailAction)
+}
+
+//MARK: - View
+
+struct RocketsListView: View {
+    
+    var store: Store<AppState, AppAction>
+    
+    init(store: Store<AppState, AppAction>) {
+        self.store = store
+    }
+    
+    var body: some View {
+        WithViewStore(self.store) { viewStore in
+            NavigationView {
+                switch viewStore.fetchingState {
+                case .loading:
+                    ProgressView()
+                    
+                case .success( _ ):
+                    ZStack {
+                        Color.ui.lightGrayList
+                        
+                        //MARK: -  Rocket info row containing:
+                        // image, rocket name, first flight
+                        
+                        List {
+                            ForEachStore(
+                                self.store.scope(
+                                    state: \.details,
+                                    action: AppAction.detail(id:action:)
+                                )
+                            ) { detailStore in
+                                WithViewStore(detailStore) { detailViewStore in
+                                    NavigationLink(destination: RocketDetailView(store: detailStore)) {
```

Comment:
Takto je to ok, ale skús modelovať aj navigáciu pomocou TCA. Teraz používaš implicitnú navigáciu, ktorú nevieš ovládať z kódu. Obecne je vždy lepšie mať čo najväčšiu kontrolu z kódu. Point free majú celú sekciu na to ako riešiť navigáciu, tak sa môžeš inšpirovať 🙂